lib/src/int64.dart

```diff
@@ -915,20 +957,20 @@ class Int64 implements IntX {
       //   - Dividing by powers of two (adjusts exponent only).
       //   - Floor (zeroes bits with fractional weight).
 
-      const double K2 = 17592186044416.0; // 2^44
-      const double K1 = 4194304.0; // 2^22
+      const K2 = 17592186044416.0; // 2^44
+      const K1 = 4194304.0; // 2^22
 
       // Approximate double values for [a] and [b].
-      double ad = a0 + K1 * a1 + K2 * a2;
-      double bd = b0 + K1 * b1 + K2 * b2;
+      var ad = a0 + K1 * a1 + K2 * a2;
```

Comment:
I believe the inferred type in most of these is `num`, which is polymorphic in the operations, so not the same as `double`. We might have to go in later with nnbd and change this to have `as double`.

Reply:
I don't believe the lint is intended to fire if the type annotation doesn't exactly match the inferred type.

Both of these are inferred as `double` today.